packages/core/src/BaseClass.ts

```diff
@@ -73,7 +71,7 @@ export class BaseClass {
     // Will apply hooks for prehookName, hookName and posthookName
     async applyHooks(hookName: HOOKS) {
         const applyHook = (name: WRAPPED_HOOKS) => {
-            const proms = [];
+            const proms: Promise<any>[] = [];
```

Comment:
Note: instead of `any` (if it matters), you can either use a generic, or `unknown`